route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -138,44 +136,48 @@ namespace route_following_plugin
             double end_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().back()).downtrack; 
             maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, prev_mvr_end_speed, target_speed_in_lanelet, route_shortest_path.back().id(),ros::Time::now()));
         }
-        
+        ROS_DEBUG_STREAM("Maneuver plan along route successfully generated");
         return maneuvers;
     }
 
     bool RouteFollowingPlugin::plan_maneuver_cb(cav_srvs::PlanManeuversRequest &req, cav_srvs::PlanManeuversResponse &resp)
     { 
         if(latest_maneuver_plan_.empty()){
             ROS_ERROR_STREAM("A maneuver plan has not been generated");
-            return true;
+            return false;
         }
-
-        lanelet::BasicPoint2d current_loc(pose_msg_.pose.position.x, pose_msg_.pose.position.y);
         
-        double current_downtrack = wm_->routeTrackPos(current_loc).downtrack;
+        double current_downtrack = wm_->routeTrackPos(current_loc_).downtrack;
         double speed_progress = current_speed_;
         
         //update start distance of first maneuver and time for all maneuvers
         setManeuverStartDist(latest_maneuver_plan_.front(), current_downtrack);
         //Update time progress for maneuvers
         updateTimeProgress(latest_maneuver_plan_, ros::Time::now());
-        //update starting speed
         
         //Return the set of maneuvers which intersect those 15s
```

Comment:
I don't like this comment.  First, 15s should be a parameter, not hard-coded (or hard-commented).  Second, it assumes a previous conversation about some 15s time frame, which is not in the above context, so I don't know what it's talking about.